Chart enhancement as discussed in issue #352

matteoberla opened the pull request:
## 🎯 Description

Closes: #352   

## 📱 Changes

-Edited the account main chart to show year monthly balance;
-Added currency char to total account amount;
-Categories ordered by amount DESC;
-Added Year expense Y axis title;


## 🔍 Checklist for reviewers
- [ ] Code is formatted correctly
- [ ] Tests are passing
- [ ] New tests are added (if needed)
- [ ] Style matches the figma/designer requests
- Tested on:
    - [ ] iOS
    - [ ] Android

napitek commented:
Thanks for the contribution!
The categories sorted so much better to see than before, I just don't understand the use of `SplayTreeMap` (seems a bit overkill to me). It works, but since you consume the data in a `ListView`, I'd say a `Map.fromEntries` was more than fine.
There is a specific reason that escapes me?
The rest seems okay to me.

if you can solve the couple of conflicts, let's do a test and then merge it.

matteoberla commented:
@napitek 
My bad, I'm used to use that method but in this case it's not that convenient.
I've also noticed that when two or more categories have the same amount the `SplayTreeMap` collapses all in one category. I've changed the method and added a sorting by name.

napitek commented:
@matteoberla Hi, could you check the conflicts when you have time so we merge it?

matteoberla commented:
I've made all the changes almost a month ago, I thought it was all good, but I'll re check for all the conflicts tomorrow!

matteoberla commented:
Should be fixed now!
